src/screens/WalletScreen/index.js

```diff
@@ -115,33 +140,62 @@ class WalletScreen extends NavigatorComponent<Props & TestingModeProps & Actions
     this.props.navigator.push(screen('RECEIVE_MONEY_SCREEN'));
   };
 
+  showTransactions = (wallet) => {
+    this.setState({ transactionsModel: { visible: true, ethAddress: wallet.ethAddress } });
+  };
+
   onRefresh = () => {
     this.setState({ pullToRefreshInProgress: true });
     this.props.updateWalletList();
   };
 
+  transactionModalClose() {
+    this.setState({ transactionsModel: { visible: false } });
+  }
+
   render() {
     return (
       <View style={styles.screenContainer}>
         <Background />
         <FakeNavigationBar />
-
         <View style={styles.bodyContainer}>
-
           <ScreenTitle title={i18n.t('screens.wallet.title')} />
-          {this.props.wallets === null || _.isEmpty(this.props.wallets) ? <EmptyWalletScreen
-            onCreateWallet={this.createWallet}
-            onRestoreWallet={this.restoreWallet}
-          />
-            : <List
+          {this.props.wallets === null || _.isEmpty(this.props.wallets) ? (
+            <EmptyWalletScreen
+              onCreateWallet={this.createWallet}
+              onRestoreWallet={this.restoreWallet}
+            />
+          ) : (
+            <List
               wallets={this.props.wallets}
               onSendPress={this.sendMoney}
               onReceivePress={this.receiveMoney}
+              onTransactionPress={this.showTransactions}
               onRefresh={this.onRefresh}
               isRefreshing={this.state.pullToRefreshInProgress}
             />
-          }
+          )}
         </View>
+        <Modal
+          isVisible={this.state.transactionsModel.visible}
+        >
+          <View>
+            <TouchableHighlight
+              onPress={() => {
+                this.transactionModalClose();
+              }}
+            >
+              <Icon name='ios-close' style={styles.closeIcon} />
+            </TouchableHighlight>
+          </View>
+          <WebView
+            source={{
+              uri: `https://etherscan.io/address/${
```

Comment:
Interesting thing, I just realize that we should use different urls for different networks, e.g. for Rinkeby network (testing account) it should be `https://rinkeby.etherscan.io`

Reply:
@seland Any idea how do I obtain the network type in the component?

Reply:
You can use [this](https://github.com/Bit-Nation/BITNATION-Pangea-mobile/blob/develop/src/reducers/accounts.js#L151) selector to get current account from redux state. There is a field called `networkType` on account object. It can be `dev` or `main`

Reply:
@seland Added `networkType` to check which network to use.